core/ledger/pvtdatastorage/reconcile_missing_pvtdata.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package pvtdatastorage
+
+import (
+	"github.com/hyperledger/fabric-protos-go/ledger/rwset"
+	"github.com/hyperledger/fabric/common/ledger/util/leveldbhelper"
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/willf/bitset"
+)
+
+// CommitPvtDataOfOldBlocks commits the pvtData (i.e., previously missing data) of old blocks.
+// The parameter `blocksPvtData` refers a list of old block's pvtdata which are missing in the pvtstore.
+// Given a list of old block's pvtData, `CommitPvtDataOfOldBlocks` performs the following three
+// operations
+// (1) construct update entries (i.e., dataEntries, expiryEntries, missingDataEntries)
+//     from the above created data entries
+// (2) create a db update batch from the update entries
+// (3) commit the update batch to the pvtStore
+func (s *Store) CommitPvtDataOfOldBlocks(blocksPvtData map[uint64][]*ledger.TxPvtData) error {
+	if s.isLastUpdatedOldBlocksSet {
+		return &ErrIllegalCall{`The lastUpdatedOldBlocksList is set. It means that the
+		stateDB may not be in sync with the pvtStore`}
+	}
+
+	logger.Debugf("Constructing pvtdatastore entries for pvtData of [%d] old blocks", len(blocksPvtData))
+	entries, err := s.constructEntries(blocksPvtData)
+	if err != nil {
+		return err
+	}
+
+	logger.Debug("Constructing update batch from pvtdatastore entries")
+	batch := s.db.NewUpdateBatch()
+	if err := entries.addToUpdateBatch(batch); err != nil {
+		return err
+	}
+
+	logger.Debug("Committing the update batch to pvtdatastore")
+	return s.db.WriteBatch(batch, true)
+}
+
+func (s *Store) constructEntries(blocksPvtData map[uint64][]*ledger.TxPvtData) (*entriesForPvtDataOfOldBlocks, error) {
```

Comment:
`OldBlockDataProcessor` (that wraps the actual store) and have these functions (that seem generic in name but are used only in the path of old block processing) on that type? The other option could be to keep adding OldPvtdata in the function names but that makes them unnecessarily longer.

Reply:
Just to clarify, we can construct this `OldBlockDataProcessor` in the function `CommitPvtDataOfOldBlocks`

Reply:
Agree. The goal is to first complete the deprioritization of missing pvtdata. I will do larger refactoring later. I made a few changes so that I can structure the deprioritization well.

Reply:
Misunderstood your comment. What you suggested can be done in this PR, itself. Pushing an update.